src/main/java/org/springframework/data/neo4j/repository/query/QueryFragmentsAndParameters.java

```diff
@@ -73,10 +75,21 @@ public final class QueryFragmentsAndParameters {
 	@Nullable
 	private NodeDescription<?> nodeDescription;
 
+	public QueryFragmentsAndParameters(@Nullable NodeDescription<?> nodeDescription, QueryFragments queryFragments,
```

Comment:
Design-wise, using a builder could help to avoid introducing additional constructors for such a class when extending arguments. That's a learning from various other Spring Data modules that can make your changes less painful.